content/blog/adding-multi-threading-to-javascript-using-web-workers/index.md

```diff
@@ -0,0 +1,220 @@
+---
+title: "Web Workers: Adding Multi-threading to JavaScript"
+date: "2020-09-07"
+coverImage: "cover.jpg"
+author: "Hridayesh Sharma"
+tags: ["JavaScript", "Web"]
+description: "Web workers have been around for a long time and add multi-threading capability to JavaScript. Let's dive in and see how they work."
+---
+
+Once upon a time, me and my friend Hriday went to a club called **JavaScripters**. The club had some rules, that in order to enter you need to be a member of the club. There was this guy at the door who was checking for membership and if you are not a member, he would register you right there and give you a membership card. Hriday was a long time member and went in, and me, it was just my first time to any club. 😅 So, I got stuck in the registration process and so did everyone else after me. After 2-3 minutes I went in and everyone else after me.
+
+While coming back, I thought why don't these guys add one more person who would be in charge of registring people and once registred can be allowed entry, so that others don't get stuck and someone like me doesn't get yelled at for no reason.😏
+It appears that like JavaScript they are single-threaded too and don't know about web workers yet.🤔
+
+##### Well, let's teach them some basics about web workers. 😎
+
+## Web Workers: Adding Multi-threading to JavaScript
+
+### What are web workers ?
+
+_From MDN_:
+
+> Web Workers makes it possible to run a script operation in a background thread separate from the main execution thread of a web application. The advantage of this is that laborious processing can be performed in a separate thread, allowing the main (usually the UI) thread to run without being blocked/slowed down.
+
+In short, web workers let's you do multi-threading in JavaScript which is a single-threaded language.
+
+Let's create our own virtual **JavaScripters** and we are going to make sure that someone innocent doesn't gets yelled at by adding multi-threading to our JavaScript using web workers. 🤩
+
+### Let's get the party started. 🕺
+
+#### Step 1: Create a project folder and add index.html in the root of it.
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Club JavaScripters</title>
+  </head>
+  <body>
+    <button>Can you click me?</button>
+    <h2>Let's do the party. 👯‍♂️‍</h2>
+    <ul id="club"></ul>
```

Comment:
` <ul id="root"></ul>`

Reply:
updated the rest if the code for id `club`